### PR TITLE
CDAP-17482 Compute and set the sequence numbers based on the content of existing bq tables.

### PR DESCRIPTION
**Pull request by sagarkapare**
JIRA: https://cdap.atlassian.net/browse/CDAP-17482

### CODE REVIEW

File: src/main/java/io/cdap/delta/bigquery/BigQueryUtils.java [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+                                               EncryptionConfiguration encryptionConfig) throws Exception {
+    if (bigQuery.getTable(tableId) == null) {
+      return 0L;
+    }
+
+    String query = String.format("SELECT MAX(_sequence_num) FROM %s.%s", tableId.getDataset(), tableId.getTable());
+    return executeMaxQuery(bigQuery, query, encryptionConfig);
+  }
+
+  private static long executeMaxQuery(BigQuery bigQuery, String query, EncryptionConfiguration encryptionConfig)
```
Comment (seanzhougoogle) [inline src/main/java/io/cdap/delta/bigquery/BigQueryUtils.java]: nit: executeAggregateQuery

Here the function itself is not executing a MaxQuery, instead it's executing a query passed in which is expected to only generate an aggregated result (one row and one column)

File: src/main/java/io/cdap/delta/bigquery/BigQueryUtils.java [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    JobId jobId = JobId.of(UUID.randomUUID().toString());
+    Job queryJob = bigQuery.create(JobInfo.newBuilder(jobConfig).setJobId(jobId).build());
+    queryJob.waitFor();
+    TableResult result = queryJob.getQueryResults();
+    Iterator<FieldValueList> resultIter = result.iterateAll().iterator();
+    if (!resultIter.hasNext()) {
+      return 0L;
+    }
+    // query is SELECT MAX(_sequence_num) FROM ...
+    // so there is at most one row and one column in the output.
```
Comment (seanzhougoogle) [inline src/main/java/io/cdap/delta/bigquery/BigQueryUtils.java]: nit:
// the query passed in is expected to generate an aggregated result which contains one row and one column.

File: src/main/java/io/cdap/delta/bigquery/BigQueryUtils.java [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+    builder.append("SELECT MAX(max_sequence_num) FROM (");
+    Set<SourceTable> allTables = context.getAllTables();
+    List<String> maxSequenceNumQueryPerTable = new ArrayList<>();
+    for (SourceTable table : allTables) {
+      TableId tableId = TableId.of(project,
+                                   datasetName != null ? normalize(datasetName) : normalize(table.getDatabase()),
+                                   normalize(table.getTable()));
+      if (bigQuery.getTable(tableId) != null) {
+        maxSequenceNumQueryPerTable.add(String.format("SELECT MAX(_sequence_num) as max_sequence_num FROM %s.%s.%s",
+                                                      tableId.getProject(), tableId.getDataset(), tableId.getTable()));
```
Comment (seanzhougoogle) [inline src/main/java/io/cdap/delta/bigquery/BigQueryUtils.java]: do we need the project ? getMaximumSequenceNumberForTable does't use project, shall we make them consistent?

**Review by seanzhougoogle — APPROVED** [review]
Leave some minor comments otherwise LGTM :)

File: src/main/java/io/cdap/delta/bigquery/BigQueryUtils.java (hunk 1 above)
Comment (sagarkapare) [reply src/main/java/io/cdap/delta/bigquery/BigQueryUtils.java]: +1 fixed.

File: src/main/java/io/cdap/delta/bigquery/BigQueryUtils.java (hunk 2 above)
Comment (sagarkapare) [reply src/main/java/io/cdap/delta/bigquery/BigQueryUtils.java]: fixed.

File: src/main/java/io/cdap/delta/bigquery/BigQueryUtils.java (hunk 3 above)
Comment (sagarkapare) [reply src/main/java/io/cdap/delta/bigquery/BigQueryUtils.java]: Its not required. Removed.

**Comment by sagarkapare** [comment]
https://builds.cask.co/browse/DEL-BPBT95-4 Squashing and merging.

**Pull request merged**